jvm/src/main/scala/fm/common/ClassUtil.scala

```diff
@@ -212,16 +218,17 @@ object ClassUtil extends Logging {
     val urls: Set[URL] = classLoader.getResources(packageDirPath).asScala.toSet
     
     urls.flatMap { url: URL =>
+      val filePath: String = URLDecoder.decode(url.getFile(), "UTF-8")
+
       url.getProtocol() match {
-        case "jar" => 
-          val fullJarUrl: String = URLDecoder.decode(url.getFile(), "UTF-8")
-          val jarFile: String = fullJarUrl.substring("file:".length, fullJarUrl.indexOf("!"))
-          val jarPrefix: String = fullJarUrl.substring(fullJarUrl.indexOf("!") + 1)
-          require(jarPrefix == File.separator+packageDirPath, s"Expected jarPrefix ($jarPrefix) to equal package prefix ($packageDirPath)")
-          scanJar(packageDirPath+File.separator, new File(jarFile))
+        case "jar" =>
```

Comment:
They currently *don't*, but I didn't change the code behavior for the `"jar"` path, i just factored out the `val fullJarUrl: String = URLDecoder.decode(url.getFile(), "UTF-8")` into  `val filePath: String = URLDecoder.decode(url.getFile(), "UTF-8")` above.  I'll look into figuring out if there is a good test to do here based upon our current dependencies.